src/A4_create_pcts_master.py

```diff
@@ -63,31 +65,97 @@ def find_parents():
+    df1 = (pd.get_dummies(df.prefix.apply(pd.Series).stack()).sum(level=0)
+        .fillna(0)
+        .drop(columns = "ZAI")
+        )
 
+    # ZAI appears in prefixes..but is not a valid prefix, it's actually a suffix
+    new = pd.merge(df, df1, left_index = True, right_index = True)
+    
     # Turn the list of suffixes into dummies
-    df2 = pd.get_dummies(df.suffix.apply(pd.Series).stack()).sum(level=0).fillna(0)
-    new = pd.merge(new, df2, left_index = True, right_index = True)   
+    df2 = (pd.get_dummies(df.suffix.apply(pd.Series).stack()).sum(level=0)
+        .fillna(0)
+        .drop(columns = "")
+        )
+    new2 = pd.merge(new, df2, left_index = True, right_index = True) 
     
-    parents = (new.drop(columns = ['CASE_ID', 'CASE_NBR', 'prefix', 'invalid', 'suffix'])
-                .pivot_table(index = ['PARENT_CASE'], aggfunc = 'max')
-                .reset_index()
+
+    # Resolve differences. 
+    # A couple of suffixes appear in both prefix and suffix list.
+    # This is because the suffix does appear in the prefix position.
+    # We want to capture this info once and correctly move it into a suffix dummy. 
+    appears_twice = ["CUB", "EIR", "CUZ", ]
+
+    def grab_max(df, col):
+        col_x = f"{col}_x"
+        col_y = f"{col}_y"
+        
+        group_together = [col_x, col_y]
+        
+        new_name = f"{col}"
+        
+        df = (df.assign(
+            new = df[group_together].max(axis=1)
+            ).rename(columns = {"new": new_name})
+            .drop(columns = group_together)
+        )
+        
+        return df
+
+    for c in appears_twice:
+        new2 = grab_max(new2, c)
+
+
+    # Reorder columns
+    prefix_order = ['PARENT_CASE', 'AA', 'ADM', 'APCC', 'APCE', 'APCH', 'APCNV',
+        'APCS', 'APCSV', 'APCW', 'CHC', 'CPC', 'DIR', 'ENV', 'PAR', 'PM', 'ZA']
+    
+    suffix_order = ['1A', '2A', 'AC', 'ACI', 'ADD1', 'ADU', 'AIC', 
+        'BL', 'BSA', 'CA', 'CASP', 'CATEX', 'CC', 'CC1', 'CC3', 'CCMP', 'CDO', 
+        'CDP', 'CE', 'CEX', 'CLQ', 'CM', 'CN', 'COA', 'COC', 'CPIO', 'CPIOA', 
+        'CPIOC', 'CPIOE', 'CPU', 'CR', 'CRA', 'CU', 'CUB', 'CUC', 'CUE', 'CUW', 'CUX', 'CUZ', 
+        'CWC', 'CWNC', 'DA', 'DB', 'DD', 'DEM', 'DI', 'DPS', 'DRB', 'EAF', 'EIR', 'ELD', 
+        'EXT', 'EXT2', 'EXT3', 'EXT4', 'F', 'GB', 'GPA', 'GPAJ', 'HCA', 'HCM', 'HD', 'HPOZ', 
+        'ICO', 'INT', 'M1', 'M2', 'M3', 'M6', 'M7', 'M8', 'M9', 'M10', 'M11', 'MA', 'MAEX',
+        'MCUP', 'MEL', 'MND', 'MPA', 'MPC', 'MPR', 'MSC', 'MSP', 'NC', 'ND', 'NR',
+        'O', 'OVR', 'P', 'PA', 'PA1', 'PA2', 'PA3', 'PA4', 'PA5', 'PA6', 'PA7', 'PA9', 'PA10',
+        'PA15', 'PA16', 'PA17', 'PAB', 'PAD', 'PMEX', 'PMLA', 'PMW', 'POD', 
+        'PP', 'PPR', 'PPSP', 'PSH', 'PUB', 'QC', 'RAO', 'RDP', 'RDPA',
+        'REC1', 'REC2', 'REC3', 'REC4', 'REC5', 'REV', 'RFA', 'RV',
+        'SCEA', 'SCPE', 'SE', 'SIP', 'SL', 'SLD', 'SM', 'SN', 'SP', 'SPE', 'SPP', 'SPPA', 'SPPM', 

```

Comment:
Can't we just sort the prefix/suffix lists?

Reply:
Yeah, I was being finicky and preferential about the "PA1", "PA2", ..., "PA10", instead of "PA1", "PA10", "PA2", but I can overlook that if it's generally alphabetically sorted. 

New version will be much cleaner in just appending alphabetically sorted prefix_list and suffix_list to act as the new column order.